contracts/FlowStakingCollection.cdc

```diff
@@ -392,11 +402,26 @@ access(all) contract FlowStakingCollection {
         /// Operations to register new staking objects
 
         /// Function to register a new Staking Record to the Staking Collection
-        access(CollectionOwner) fun registerNode(id: String, role: UInt8, networkingAddress: String, networkingKey: String, stakingKey: String, amount: UFix64, payer: AuthAccount): AuthAccount? {
+        access(CollectionOwner) fun registerNode(
+            id: String,
+            role: UInt8,
+            networkingAddress: String,
+            networkingKey: String,
+            stakingKey: String,
+            amount: UFix64,
+            payer: auth(BorrowValue) &Account
+        ): auth(Storage, Capabilities, Contracts, Keys, Inbox) &Account? {
```

Comment:
Is it still necessary to return a fully authorized account reference here? Maybe reduce the entitlements here to just what is needed, if possible

Reply:
The returned account is used by various transactions for different purposes (there were some which add keys, etc.). I don't quite know what are all the use cases (depends on the business logic). 

So, for now, I updated this in a way that it makes it easy to update the downstream deps, and most closest to the original code. We would need to do a second round of review/update to see if the provided entitlement (this and in other places), are actually enough/too-much.

Reply:
Sounds good! Maybe open a "follow-up issue" to suggest tightening the returned type to just what is needed

Reply:
Done: https://github.com/onflow/flow-core-contracts/issues/386